onvm/onvm_mgr/onvm_stats.c

```diff
@@ -489,27 +489,27 @@ onvm_stats_display_nfs(unsigned difftime, uint8_t verbosity_level) {
 
                 /* Save stats for NFs with same service id */
                 if (print_total_stats) {
-                        rx_for_service[nfs[i].info->service_id] += rx;
-                        tx_for_service[nfs[i].info->service_id] += tx;
-                        rx_drop_for_service[nfs[i].info->service_id] += rx_drop;
-                        tx_drop_for_service[nfs[i].info->service_id] += tx_drop;
-                        rx_pps_for_service[nfs[i].info->service_id] += rx_pps;
-                        tx_pps_for_service[nfs[i].info->service_id] += tx_pps;
-                        rx_drop_rate_for_service[nfs[i].info->service_id] += rx_drop_rate;
-                        tx_drop_rate_for_service[nfs[i].info->service_id] += tx_drop_rate;
-                        act_out_for_service[nfs[i].info->service_id] += act_out;
-                        act_tonf_for_service[nfs[i].info->service_id] += act_tonf;
-                        act_drop_for_service[nfs[i].info->service_id] += act_drop;
-                        act_next_for_service[nfs[i].info->service_id] += act_next;
-                        act_buffer_for_service[nfs[i].info->service_id] += act_buffer;
-                        act_returned_for_service[nfs[i].info->service_id] += act_returned;
+                        rx_for_service[nfs[i].service_id] += rx;
+                        tx_for_service[nfs[i].service_id] += tx;
+                        rx_drop_for_service[nfs[i].service_id] += rx_drop;
+                        tx_drop_for_service[nfs[i].service_id] += tx_drop;
+                        rx_pps_for_service[nfs[i].service_id] += rx_pps;
+                        tx_pps_for_service[nfs[i].service_id] += tx_pps;
+                        rx_drop_rate_for_service[nfs[i].service_id] += rx_drop_rate;
+                        tx_drop_rate_for_service[nfs[i].service_id] += tx_drop_rate;
+                        act_out_for_service[nfs[i].service_id] += act_out;
+                        act_tonf_for_service[nfs[i].service_id] += act_tonf;
+                        act_drop_for_service[nfs[i].service_id] += act_drop;
+                        act_next_for_service[nfs[i].service_id] += act_next;
+                        act_buffer_for_service[nfs[i].service_id] += act_buffer;
+                        act_returned_for_service[nfs[i].service_id] += act_returned;
                 }
 
                 if (verbosity_level == ONVM_RAW_STATS_DUMP) {
                         fprintf(stats_out, "%s,%u,%u,%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64
                                            ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64
                                            ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ",%" PRIu64 ", %d\n",
-                                buffer, nfs[i].info->instance_id, nfs[i].info->service_id, rx, tx, rx_pps, tx_pps,
+                                buffer, nfs[i].instance_id, nfs[i].service_id, rx, tx, rx_pps, tx_pps,
```

Comment:
I just realized unfortunately, that my new stats code adds more references to nf_info, instead of your new struct. Can we wait on merging this until `update_stats` is finished?

Reply:
Yeah I'm aware of this, don't worry depending on what gets merged first we'll resolve the conflicts

Reply:
Great